Hotfix: Cleaning up feature flags + banner cards (#8604)
braze-inc/braze-docs: _docs/_developer_guide/feature_flags.md

---
nav_title: Feature Flags
article_title: Feature flags for the Braze SDK
page_order: 1
description: "This reference article covers an overview of feature flags including prerequisites and use cases."
tool: Feature Flags
platform:
  - iOS
  - Android
  - Web
  - Unity
  - Cordova
  - React Native
  - Flutter
  - Roku

---

# Feature flags

> Feature flags allow you to remotely enable or disable functionality for a specific or random selection of users. Importantly, they let you turn a feature on and off in production without additional code deployment or app store updates. This allows you to safely roll out new features with confidence.

{% alert tip %}

> When you're ready to create your own feature flags, checkout [Creating feature flags]({{site.baseurl}}/developer_guide/platform_wide/feature_flags/create/).

Reviewer: ```suggestion
When you're ready to create your own feature flags, check out [Creating feature flags]({{site.baseurl}}/developer_guide/platform_wide/feature_flags/create/).
```

For example, let's say we've decided to add a new "Live Chat Support" link to our app for faster customer service. We could release this feature to all customers at once. However, a wide release carries risks, such as: 

* Our Support team is still in training, and customers can start support tickets after it's released. This doesn't give us any leeway in case the Support team needs more time.
* We're unsure of the actual volume of new support cases we'll get, so we might not be staffed appropriately.
* If our Support team is overwhelmed, we have no strategy to quickly turn this feature off again.
* There might be bugs introduced in the chat widget, and we don't want customers to have a negative experience.

With Braze feature flags, we can instead gradually roll out the feature and mitigate all of these risks:

* We will turn on the "Live Chat Support" feature when the Support team says they're ready.
* We will enable this new feature for only 10% of users to determine if we're staffed appropriately.
* If there are any bugs, we can quickly disable the feature instead of rushing to ship a new release.

> To gradually roll out this feature, we can [create]({{site.baseurl}}/developer_guide/platform_wide/feature_flags/create/) a feature flag named "Live Chat Widget."

Reviewer: Minor nit that this link text isn't long enough to be descriptive. Added a suggested revision! 

```suggestion
To gradually roll out this feature, we can [create a feature flag]({{site.baseurl}}/developer_guide/platform_wide/feature_flags/create/) named "Live Chat Widget."
```

// Read the "link" property
const [promoLink, setPromoLink] = useState(featureFlag.getStringProperty("link"));
// Read the "text" property
const [promoText, setPromoText] = useState(featureFlag.getStringProperty("text"));

return (<>
  <div>
    <a href="/">Home</a>
    { promoEnabled && <a href={promoLink}>{promoText}</a> }
    <a href="/products">Products</a>
    <a href="/categories">Categories
  </div>
</>)
```

Now, the day before Thanksgiving, we only have to change those property values in the Braze dashboard.

![Feature flag with link and text properties directing to a Thanksgiving sales page.]({% image_buster /assets/img/feature_flags/feature-flags-use-case-navigation-link-2.png %})

As a result, the next time someone loads the app, they will see the new Thanksgiving deals.

### Message coordination

Use feature flags to synchronize a feature's rollout and messaging. This will allow you to use Braze as the source of truth for both your user experience and its relevant messaging. To achieve this, target the new feature to a particular segment or filtered portion of your audience. Then, create a campaign or Canvas that only targets that segment. 

> Let's say that we're launching a new loyalty rewards program for our users. It can be difficult for Marketing and Product teams to perfectly coordinate the timing of promotional messaging with a feature's rollout. Feature flags in Canvas let you apply sophisticated logic when it comes to enabling a feature for a select audience and controlling the related messaging to those same users.

Reviewer: Minor inconsistency that "marketing team" is not capitalized above, but is here.

```suggestion
Let's say that we're launching a new loyalty rewards program for our users. It can be difficult for marketing and product teams to perfectly coordinate the timing of promotional messaging with a feature's rollout. Feature flags in Canvas let you apply sophisticated logic when it comes to enabling a feature for a select audience and controlling the related messaging to those same users.
```
